### ISSUE TITLE
[Mobile] I get an error that nothing but strings are allowed, even though they are strings.

### ISSUE THREAD

**Issue by mono0218** [microsoft/onnxruntime#16301]
### Describe the issue

I get an error that nothing but strings are allowed, even though they are strings.

`import {Env, Tensor} from "onnxruntime-react-native";
import * as ort from "onnxruntime-react-native";
import RNFS from "react-native-fs";
import {Image} from 'react-native';
export async function post(message){
    const dest = RNFS.DocumentDirectoryPath+"/monokamo.ort";
    const sess = await ort.InferenceSession.create("file://"+dest)
    const input = new Tensor("string",[message],[1,1])
    const result = await sess.run({text: input});
    console.log(result)
}`

### To reproduce

Run the code pasted above

### Platform

React Native

### OS Version

Pixel_3a_API_33_arm64-v8a(AVD)

### ONNX Runtime Installation

Released Package

### Package Name (if 'Released Package')

None

### ONNX Runtime Version or Commit ID

1.15.0

### ONNX Runtime API

JavaScript

### Architecture

X64

### Execution Provider

Default CPU

**Comment by YUNQIUGUO** [comment]
FYI: Looks like the error message comes from here: 

https://github.com/microsoft/onnxruntime/blob/c8bd34f9750ebc0cbd2443a90ad06a647ea5c203/js/react_native/android/src/main/java/ai/onnxruntime/reactnative/OnnxruntimeModule.java#L227-L238

You may want to provide the input as a base64 encoded type string for now.

Here's an example usage in react_native/e2e test app as a reference: (in constructing a Float32Array type though) 

https://github.com/microsoft/onnxruntime/blob/2cb3fb18b5f168550ccfb607286b4f42750a76e6/js/react_native/e2e/src/App.tsx#L58-L69

**Comment by fs-eire** [comment]
This is not correct. The actual reason is that our Java code does not implement for string tensors.

The data is serialized to a single BASE64 strings for non-string tensors, and will pass as a string array for string tensors.
https://github.com/microsoft/onnxruntime/blob/2cb3fb18b5f168550ccfb607286b4f42750a76e6/js/react_native/lib/backend.ts#L112-L118

To support string tensors, need to add code to handle them in OnnxruntimeModule.java

**Comment by fs-eire** [comment]
> FYI: Looks like the error message comes from here:
> 
> https://github.com/microsoft/onnxruntime/blob/c8bd34f9750ebc0cbd2443a90ad06a647ea5c203/js/react_native/android/src/main/java/ai/onnxruntime/reactnative/OnnxruntimeModule.java#L227-L238
> 
> You may want to provide the input as a base64 encoded type string for now.
> 
> Here's an example usage in react_native/e2e test app as a reference: (in constructing a Float32Array type though)
> 
> https://github.com/microsoft/onnxruntime/blob/2cb3fb18b5f168550ccfb607286b4f42750a76e6/js/react_native/e2e/src/App.tsx#L58-L69

please don't refer to this example. The e2e test uses Java code to generate BASE64 encoded string for the data (`mnistInput[key].data`) and the JavaScript code tries to decode it to generate a raw buffer.

For normal users who consumes `onnxruntime-react-native` package, the BASE64 encoding/decoding is transparent to users and we do not expect users to write code in this way.

Users should write code normally, for example:

```js
const tensorData = Float32Array.from([1.0, 2.0, 3.0, 4.0]);
const myTensor = new Tensor('float32', tensorData, [2, 2]);
```

**Comment by YUNQIUGUO** [comment]
ah okay. I see. sorry about the confusion. I deleted the referenced e2e example.

**Comment by mono0218** [comment]
> > FYI: Looks like the error message comes from here:
> > https://github.com/microsoft/onnxruntime/blob/c8bd34f9750ebc0cbd2443a90ad06a647ea5c203/js/react_native/android/src/main/java/ai/onnxruntime/reactnative/OnnxruntimeModule.java#L227-L238
> > 
> > You may want to provide the input as a base64 encoded type string for now.
> > Here's an example usage in react_native/e2e test app as a reference: (in constructing a Float32Array type though)
> > https://github.com/microsoft/onnxruntime/blob/2cb3fb18b5f168550ccfb607286b4f42750a76e6/js/react_native/e2e/src/App.tsx#L58-L69
> 
> please don't refer to this example. The e2e test uses Java code to generate BASE64 encoded string for the data (`mnistInput[key].data`) and the JavaScript code tries to decode it to generate a raw buffer.
> 
> For normal users who consumes `onnxruntime-react-native` package, the BASE64 encoding/decoding is transparent to users and we do not expect users to write code in this way.
> 
> Users should write code normally, for example:
> 
> ```js
> const tensorData = Float32Array.from([1.0, 2.0, 3.0, 4.0]);
> const myTensor = new Tensor('float32', tensorData, [2, 2]);
> ```
> 
> The only reason why the error "Error: Non string type of a tensor data is not allowed" happen is because string tensor support is not implemented in OnnxruntimeModule.java. The error message is a little bit confusing.

Thanks for the reply.
I was able to infer successfully by deleting the if statement from line 227 of OnnxruntimeModule.java.

This means that I can infer even with strings.
Perhaps the if statement needs to allow arrays as well!

**Comment by fs-eire** [comment]
thank you for the verification. I didn't realize that creating string tensor is already implemented in https://github.com/microsoft/onnxruntime/blob/c8bd34f9750ebc0cbd2443a90ad06a647ea5c203/js/react_native/android/src/main/java/ai/onnxruntime/reactnative/TensorHelper.java#L63-L69

yes, we need to change this line to support Array type as well.
```
if (inputMap.getType("data") != ReadableType.String)
```

**Comment by mono0218** [comment]
Pull request submitted.[URL](https://github.com/microsoft/onnxruntime/pull/16348)
